Description: Don't require FutureWarning to be an error

setup.cfg tries to make it one, but this doesn't work in our CI setup

Author: Rebecca N. Palmer
Forwarded: no

--- a/dask/array/tests/test_linalg.py
+++ b/dask/array/tests/test_linalg.py
@@ -806,12 +806,12 @@
     assert_eq(res, _scipy_linalg_solve(A, b, assume_a="pos"), check_graph=False)
     assert_eq(dA.dot(res), b.astype(float), check_graph=False)
 
-    with pytest.raises(FutureWarning, match="sym_pos keyword is deprecated"):
+    with pytest.warns(FutureWarning, match="sym_pos keyword is deprecated"):
         res = da.linalg.solve(dA, db, sym_pos=True)
         assert_eq(res, _scipy_linalg_solve(A, b, assume_a="pos"), check_graph=False)
         assert_eq(dA.dot(res), b.astype(float), check_graph=False)
 
-    with pytest.raises(FutureWarning, match="sym_pos keyword is deprecated"):
+    with pytest.warns(FutureWarning, match="sym_pos keyword is deprecated"):
         res = da.linalg.solve(dA, db, sym_pos=False)
         assert_eq(res, _scipy_linalg_solve(A, b, assume_a="gen"), check_graph=False)
         assert_eq(dA.dot(res), b.astype(float), check_graph=False)
